include/nbl/video/IGPUSpecializedShader.h

```diff
@@ -22,17 +22,16 @@ namespace video
 class IGPUSpecializedShader : public asset::ISpecializedShader, public IBackendObject
 {
 	public:
-		IGPUSpecializedShader(ILogicalDevice* dev, asset::ISpecializedShader::E_SHADER_STAGE _stage) : IBackendObject(dev), m_stage(_stage) {}
+		IGPUSpecializedShader(core::smart_refctd_ptr<const ILogicalDevice>&& dev, asset::IShader::E_SHADER_STAGE _stage) : IBackendObject(std::move(dev)), m_stage(_stage) {}
```

Comment:
> why is the constructor public

Always has been, but seems like it can be made `protected` without breaking anything, do  you want me to do it?

>  why does it take the stage!?

Always has been, how else `COpenGLSpecializedShader` would know the stage because it doesn't hold a refctd to pointer to unspecialized shader like `CVulkanSpecializedShader` does?

Reply:
> Always has been, how else COpenGLSpecializedShader would know the stage because it doesn't hold a refctd to pointer to unspecialized shader like CVulkanSpecializedShader does?

No, not true, removing the `asset::IShader::E_SHADER_STAGE _stage` param.

Reply:
Resolved